third_party/move/move-compiler-v2/src/pipeline/livevar_analysis_processor.rs

```diff
@@ -141,12 +155,57 @@ impl LiveVarAnalysisProcessor {
             code,
             &cfg,
         );
-        analyzer.state_per_instruction(state_map, code, &cfg, |before, after| {
-            LiveVarInfoAtCodeOffset {
-                before: before.livevars.clone().into_iter().collect(),
-                after: after.livevars.clone().into_iter().collect(),
+        // Prepare the result as a map from CodeOffset to LiveVarInfo
+        let mut code_map =
+            analyzer.state_per_instruction(state_map, code, &cfg, |before, after| {
+                LiveVarInfoAtCodeOffset {
+                    before: before.livevars.clone().into_iter().collect(),
+                    after: after.livevars.clone().into_iter().collect(),
+                }
+            });
+
+        // Now propagate to all branches in the code the after set of the branch instruction. Consider code as follows:
+        // ```
+        // L0: if c goto L1 else L2
+        // L1: ..
+        //     goto L0
+        // L2: ..
+        // ```
+        // The backwards analysis will not populate the before state of L2, which should be the same as the after
```

Comment:
"After state of L0 must be the same as before state of L2": I am not sure if we can expect this statement to hold from backwards dataflow analysis, whose fixed point equations only say that "after(B) is the union of before(S), where S is a successor of B" (that is, after(L2) should only be influenced by its successors, and before(L2) should be influenced by after(L2) and the instruction at L2).

Reply:
I think I do understand the argument for why we do not need a fixpoint and can do this propagation better. Added to the code:
```
        // Thereby we assert that there is no fixpoint needed for this post-analysis step. The argument is as follows:
        // any live variables which we propagate from the branch source to the target have either been (a) not used
        // by the target, then they will immediately be released (b) have been used already in which case the join
        // does not make a difference.
```

Reply:
it's too late for me to analyze this, but the shape of your argument is reminding me of some work by Dhamdhere (maybe with Khedker) back in the early 90s (can't find a great citation now) about how certain analysis look bidirectional but can be turned into directional + one step in the opposite direction.  Too much notation and archaic jargon to be worth reading about, probably.  But tomorrow I'll try to understand your code better.

Reply:
I'm talking about the resulting annotation state of the program when I mean before and after, and not about the state the dataflow analysis computes. The later I see as an intermediate step to compute the final former form. Mentioning fixpoint led to some confusion, I think. See also later reply.